cmd/ingress-operator/serve-healthcheck.go

```diff
@@ -0,0 +1,72 @@
+package main
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+
+	"github.com/spf13/cobra"
+
+	canarycontroller "github.com/openshift/cluster-ingress-operator/pkg/operator/controller/canary"
+)
+
+func healthCheckHandler(w http.ResponseWriter, r *http.Request) {
+	response := os.Getenv("RESPONSE")
+	if len(response) == 0 {
+		response = canarycontroller.CanaryHealthcheckResponse
+	}
+
+	// Echo back the port the request was received on
+	// via a "request-port" header.
+	addr := r.Context().Value(http.LocalAddrContextKey).(net.Addr)
+	if tcpAddr, ok := addr.(*net.TCPAddr); ok {
+		w.Header().Set("x-request-port", strconv.Itoa(tcpAddr.Port))
+	}
+
+	_, err := fmt.Fprintln(w, response)
+	if err == nil {
+		fmt.Println("Servicing request.")
+	} else {
+		log.Error(err, "could not serve health check")
```

Comment:
Hm, if we can use the logger for errors, can we use it also (`log.Info`) for success?

Reply:
Changed to use fmt.Println for both to be consistent with a web server pattern.